ros2node/ros2node/api/__init__.py

```diff
@@ -37,8 +37,8 @@ def parse_node_name(full_node_name):
     namespace = '/'
     if len(tokens) > 1:
         node_name = tokens[-1]
-        namespace = '/'.join(tokens[:-1])
-    return NodeName(node_name, namespace, full_node_name)
+        namespace = '/' + '/'.join(tokens[:-1])
+    return NodeName(node_name, namespace, '/' + full_node_name)
```

Comment:
The terminology seems to be all mixed in this function. The parameter `full_node_name` must explicitly not have a leading slash. But the returned value of `NodeName.full_name` explicitly does.

While not part of this patch the condition `1 > len(tokens)` can never be true. Also some callers explicitly remove the leading slash before calling the function.

Reply:
> * Change callers to not remove the leading slash.
> * Also fix #181 here, adding a leading slash to non fully qualified names (inside parse_node_name).
> * Change "parse_node_name(full_node_name)" to "parse_node_name(node_name)" ...

:+1: 

> ... and node_name to "bare_node_name" or something similar.

Maybe `node_basename` to match terminology for paths?